Fix LightningCLI docs after overhaul of the documentation (#14976)
Lightning-AI/lightning: docs/source-pytorch/cli/lightning_cli_advanced_3.rst

Models with multiple submodules
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

> Many use cases require to have several modules, each with its own configurable options. One possible way to handle this
> with ``LightningCLI`` is to implement a single module having as init parameters each of the submodules. This is known as
> `dependency injection <https://en.wikipedia.org/wiki/Dependency_injection>`__ which is a good approach to improve
> decoupling in your code base.
> 
> Since the init parameters of the model have as type hint a class, in the configuration these would be specified with
> ``class_path`` and ``init_args`` entries. For instance a model could be implemented as:

Reviewer: ```suggestion
Many use cases require to have several modules, each with its own configurable options. One possible way to handle this
with ``LightningCLI`` is to implement a single module having as init parameters each of the submodules. This is known as
`dependency injection <https://en.wikipedia.org/wiki/Dependency_injection>`__ which is a good approach to improve
decoupling in your code base.

Since the init parameters of the model have as a type hint a class, in the configuration, these would be specified with
``class_path`` and ``init_args`` entries. For instance, a model could be implemented as:
```

        init_args:
          ...

It is also possible to combine :code:`subclass_mode_model=True` and submodules, thereby having two levels of
:code:`class_path`.

Class type defaults

> ^^^^^^^^^^^^^^^^^^^

Reviewer 2: I couldn't find where this was moved or why it must be removed. Could you guide the reviewer here :)

Author: I did think about adding comments to guide, but then I forgot. Sorry!

This section was duplicated by mistake, see [lightning_cli_expert.rst#class-type-defaults](https://github.com/Lightning-AI/lightning/blob/master/docs/source-pytorch/cli/lightning_cli_expert.rst#class-type-defaults)

> In some case it might be desired to fix the optimizer and/or learning scheduler scheduler instead of allowing multiple.
> For this you can manually add the arguments for specific classes by subclassing the CLI. The following code snippet
> shows how to implement it:

Reviewer: ```suggestion
In some cases, fixing the optimizer and/or learning scheduler might be desired instead of allowing multiple.
For this, you can manually add the arguments for specific classes by subclassing the CLI. The following code snippet
shows how to implement it:
```

    trainer:
      ...

> where the arguments can be passed directly through command line without specifying the class. For example:

Reviewer: ```suggestion
where the arguments can be passed directly through the command line without specifying the class. For example:
```

    cli = MyLightningCLI(MyModel)

> The value given to ``optimizer*_init`` will always be a dictionary including ``class_path`` and ``init_args`` entries.
> The function :func:`~pytorch_lightning.cli.instantiate_class` takes care of importing the class defined in
> ``class_path`` and instantiating it using some positional arguments, in this case ``self.parameters()``, and the
> ``init_args``. Any number of optimizers and learning rate schedulers can be added when using ``link_to``.

Reviewer: ```suggestion
The value given to ``optimizer*_init`` will always be a dictionary including ``class_path`` and ``init_args`` entries.
The function :func:`~pytorch_lightning.cli.instantiate_class` takes care of importing the class defined in
``class_path`` and instantiating it using some positional arguments, in this case ``self.parameters()``, and the
``init_args``. Any number of optimizers and learning rate schedulers can be added when using ``link_to``.
```
